### ISSUE TITLE
`useModal` hook, empty props

### ISSUE THREAD

**Issue by gino** [mpontus/react-modal-hook#50]
Hi there,

I am trying to pass the actual modal "open" state to my modal and expected an `in` prop that should contain a boolean.
Unfortunately, I tried to use this `in` prop and didn't get anything back.

```tsx
const [showModal, hideModal] = useModal((params) => {
  // Params = {}
  console.log(params);

  return <div>My modal</div>;
});
```

However, I figured out that the `params` here, is an empty object.. anyone that could help me figure out why I am not receiving the ‘open’ boolean in here? I might be missing something. The modal works fine though, so it's quite strange. I am using Next.js.

Thanks!

**Comment by vitorcamachoo** [comment]
I have this problem too. Any way to fix ?

**Comment by darrist** [comment]
Looks like these are callback from the modal component, so if your self-rolled modal component doesn't callback then you will not get any of these from `useModal((params) =>`.

What could be nice is if repo owner will add open state to the lib.

**Comment by gino** [comment]
I have fixed this issue, which was not an issue related to Next.js.

Apparently it is required to pass the `rootComponent` prop to the `ModalProvider` in order to have the `in` parameter passed in. So I passed in the TransitionGroup as specified in [here](https://github.com/mpontus/react-modal-hook#animated-modals) and now I do have the "in" prop. It also makes sense now if I read the "Animated modals" part again..
